examples/silentpayments.c

```diff
@@ -0,0 +1,543 @@
+        sp_addresses[2] = &carol_address; /* : 3.0 BTC */
+        for (i = 0; i < N_OUTPUTS; i++) { ret =
+            secp256k1_ec_pubkey_parse(ctx,
+                &recipients[i].scan_pubkey,
+                (*(sp_addresses[i]))[0],
+                33
+            );
+            assert(ret);
+            ret = secp256k1_ec_pubkey_parse(ctx,
+                &recipients[i].spend_pubkey,
+                (*(sp_addresses[i]))[1],
+                33
+            );
+            assert(ret);
+            /* Alice creates the recipient objects and adds the index of the
+             * original ordering (the ordering of the `sp_addresses` array) to
+             * each object. This index is used to return the generated outputs
+             * in the original ordering so that Alice can match up the generated
+             * outputs with the correct amounts.
+             */
+            recipients[i].index = i;
+            recipient_ptrs[i] = &recipients[i];
+        }
+        for (i = 0; i < N_OUTPUTS; i++) {
+            tx_output_ptrs[i] = &tx_outputs[i];
+        }
+        /* To keep things simple, we cast the tx_output_ptr array to remove the
+         * const qualifer, so that we can create the outputs. We want the const
+         * qualifer because this same array will be passed to the scan function
+         * later in the example.
+         */
+        ret = secp256k1_silentpayments_sender_create_outputs(ctx,
+            (secp256k1_xonly_pubkey **)tx_output_ptrs,
+            recipient_ptrs, N_OUTPUTS,
+            smallest_outpoint,
+            sender_keypair_ptrs, N_INPUTS,
+            NULL, 0
+        );
+        assert(ret);
+        printf("Alice created the following outputs for Bob and Carol: \n");
+        for (i = 0; i < N_OUTPUTS; i++) {
+            printf("    ");
+            printf("%s : ", address_amounts[i]);
+            secp256k1_xonly_pubkey_serialize(ctx,
+                xonly_print,
+                &tx_outputs[i]
+            );
+            print_hex(xonly_print, sizeof(xonly_print));
+        }
+        /* It's best practice to try to clear secrets from memory after using
+         * them. This is done because some bugs can allow an attacker to leak
+         * memory, for example through "out of bounds" array access (see
+         * Heartbleed), or the OS swapping them to disk. Hence, we overwrite the
+         * secret key buffer with zeros.
+         *
+         * Here we are preventing these writes from being optimized out, as any
+         * good compiler will remove any writes that aren't used. */
+        secure_erase(seckey, sizeof(seckey));
+        for (i = 0; i < N_INPUTS; i++) {
+            secure_erase(&sender_keypairs[i], sizeof(sender_keypairs[i]));
+        }
+    }
+
+    /*** Receiving ***/
+    {
+        /*** Transaction data ***
+         *
+         * Here we create a few global variables to represent the transaction
+         * data:
```

Comment:
A bit confusing because we don't create the global variables here, we just set them. Not sure if this comment is outdated.